app/utils/component-utils.ts

```diff
@@ -0,0 +1,14 @@
+import { SimpleChange } from "@angular/core";
+
+export class ComponentUtils {
+    /**
+     * Return true if the record changed id
+     * @param change Simple change from ngOnChanges
+     */
+    public static recordChangedId(change: SimpleChange): boolean {
+        if (!change) { return false; }
+        const { previousValue, currentValue } = change;
+        const same = previousValue && currentValue && previousValue.id === currentValue.id;
+        return !same;
```

Comment:
probably just need: 

return previousValue && currentValue && previousValue.id === currentValue.id;

Reply:
doing a ! here this would be the opposite